### ISSUE TITLE
Error running Flatpak on elementaryOS

### ISSUE THREAD

**Issue by sumnerevans** [sublime-music/sublime-music-python#77]
I think this probably has to do with the fact that the default cache location is not being calculated correctly.

```
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/sublime/app.py", line 554, in on_connected_server_changed
    self.state.save()
  File "/app/lib/python3.7/site-packages/sublime/state_manager.py", line 178, in save
    os.makedirs(os.path.dirname(self.state_filename), exist_ok=True)
  File "/app/lib/python3.7/site-packages/sublime/state_manager.py", line 211, in state_filename
    'state.yaml',
  File "/usr/lib/python3.7/posixpath.py", line 94, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.7/genericpath.py", line 149, in _check_arg_types
    (funcname, s.__class__.__name__)) from None
TypeError: join() argument must be str or bytes, not 'NoneType'
```

**Comment by sumnerevans** [comment]
I did some digging and this is due to `self.config.server` being null, and therefore `CacheManager.calculate_server_hash(self.config.server)` being null.

**Comment by sumnerevans** [comment]
In GitLab by @captn3m0 on Jan 25, 2020, 07:10

Facing the same issue on Arch-Linux (installed from AUR). Stack trace is identical:

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sublime/app.py", line 598, in on_connected_server_changed
    self.state.save()
  File "/usr/lib/python3.8/site-packages/sublime/state_manager.py", line 157, in save
    os.makedirs(os.path.dirname(self.state_filename), exist_ok=True)
  File "/usr/lib/python3.8/site-packages/sublime/state_manager.py", line 203, in state_filename
    return os.path.join(
  File "/usr/lib/python3.8/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.8/genericpath.py", line 152, in _check_arg_types
    raise TypeError(f'{funcname}() argument must be str, bytes, or '
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```

Test Connection to a localhost:8080 [gonic](https://github.com/sentriz/gonic) server succeeds. The crash happens on clicking the "connect" button.

**Comment by sumnerevans** [comment]
@captn3m0 thanks for the report! And thanks for checking this project out! I'll prioritize this.

**Comment by sumnerevans** [comment]
closed via commit b9a368b714ca35f9adb52772779a2763e2566bd0

**Comment by sumnerevans** [comment]
@captn3m0 This should be fixed in `v0.8.10`.

**Comment by sumnerevans** [comment]
mentioned in commit b9a368b714ca35f9adb52772779a2763e2566bd0
